Source/WebKit/Shared/Cocoa/AuxiliaryProcessCocoa.mm

```diff
@@ -343,4 +347,34 @@ static void disableDowngradeToLayoutManager()
 #endif
 }
 
+void AuxiliaryProcess::increaseFileDescriptorLimit()
+{
+    struct rlimit currentLimits = { };
+    if (int returnCode = getrlimit(RLIMIT_NOFILE, &currentLimits)) {
+        RELEASE_LOG_ERROR(Process, "Could not getrlimit(RLIMIT_NOFILE): %d", returnCode);
+        return;
+    }
+
+    int mib[] = { CTL_KERN, KERN_MAXFILESPERPROC };
+    rlim_t maxFilesPerProc = 0;
+    size_t len = sizeof(maxFilesPerProc);
+    if (int returnCode = sysctl(mib, 2, &maxFilesPerProc, &len, NULL, 0)) {
+        RELEASE_LOG_ERROR(Process, "Could not get KERN_MAXFILESPERPROC: %d", returnCode);
+        return;
+    }
+
+    // Set the fd limit to 2560, which is the magic number used by several other frameworks. The
+    // default on macOS is 256 (see `launchctl limit`).
+    struct rlimit newLimits = currentLimits;
+    newLimits.rlim_cur = std::min({ 2560llu, currentLimits.rlim_max, maxFilesPerProc });
+
+    if (newLimits.rlim_cur < currentLimits.rlim_cur) {
+        RELEASE_LOG_ERROR(Process, "Could not increase fd limit");
```

Comment:
This log is a bit confusing. It didn't try and fail; we just computed a new limit that's smaller than the old one. Maybe the logs should show the values?